src/sheet_metal_conus.py

```diff
@@ -12,7 +12,15 @@ def calc_angle_between_points(p1, p2):
         xDiff = p2[0] - p1[0]
         yDiff = p2[1] - p1[1]
         return math.degrees(math.atan2(yDiff, xDiff))
-
+def calc_dist_between_points(p1, p2):
+        xDiff = p2[0] - p1[0]
+        yDiff = p2[1] - p1[1]
+        return math.sqrt(yDiff*yDiff + xDiff*xDiff)
+def normalize(p1, p2):
+    xDiff = p2[0] - p1[0]
+    yDiff = p2[1] - p1[1]
+    magn = calc_dist_between_points(p1,p2)
+    return (xDiff/magn, yDiff/magn)
```

Comment:
These two new functions are used to calculate arrow marker lengths and direction because we have to shorten the line to get the arrow to end ON the line properly.